### PR TITLE
Add configurable deposit amounts to testutil 

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Test helper additions 

**What does this PR do? Why is it needed?**
This PR adds extra deposit helpers to allow creating deterministic deposits with custom balances. It uses any available cache of secret keys and generates any if it needs.

This is in preparation for testing partial deposits in E2E.

### CODE REVIEW

File: shared/testutil/deposits_test.go [hunk 1]
```diff
@@ -5,6 +5,10 @@ import (
 	"encoding/hex"
 	"testing"
 
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (rauljordan) [inline shared/testutil/deposits_test.go:11]:
```suggestion
```

File: shared/testutil/deposits.go [hunk 2]
```diff
@@ -57,31 +57,10 @@ func DeterministicDepositsAndKeys(numDeposits uint64) ([]*ethpb.Deposit, []bls.S
...
-			if err != nil {
-				return nil, nil, errors.Wrap(err, "could not compute domain")
-			}
-			root, err := ssz.SigningRoot(depositData)
-			if err != nil {
-				return nil, nil, errors.Wrap(err, "could not get signing root of deposit data")
-			}
-			sigRoot, err := (&pb.SigningData{ObjectRoot: root[:], Domain: domain}).HashTreeRoot()
+			balance := params.BeaconConfig().MaxEffectiveBalance
+			deposit, err := signedDeposit(secretKeys[i], publicKeys[i].Marshal(), publicKeys[i+1].Marshal(), balance)
```
Comment (rauljordan) [inline shared/testutil/deposits.go:61]: any check here that i+1 wont exceed the list size?

File: shared/testutil/deposits.go [hunk 3]
```diff
@@ -110,18 +89,129 @@ func DeterministicDepositsAndKeys(numDeposits uint64) ([]*ethpb.Deposit, []bls.S
...
+	}
+
+	deposits := make([]*ethpb.Deposit, numDeposits)
+	// Create the new deposits and add them to the trie.
+	for i := uint64(0); i < numDeposits; i++ {
+		balance := params.BeaconConfig().MaxEffectiveBalance
+		if len(balances) == int(numDeposits) {
+			balance = balances[i]
+		}
+		deposit, err := signedDeposit(secretKeys[i], publicKeys[i].Marshal(), publicKeys[i+1].Marshal(), balance)
```
Comment (rauljordan) [inline shared/testutil/deposits.go:130]: +1

File: shared/testutil/deposits.go (hunk 2 above)
Comment (0xKiwi) [reply shared/testutil/deposits.go:61]: Not required, above we generate it with a +1 as well.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**